frontend/views/containers/modals/SignUp.vue

```diff
@@ -3,190 +3,42 @@
-          const oldSettings = await sbp('gi.db/settings/load', this.form.name)
-          if (oldSettings) {
-            // TODO: prompt to ask user before deleting and overwriting an existing user
-            //       https://github.com/okTurtles/group-income-simple/issues/599
-            console.warn(`deleting settings for pre-existing user ${this.form.name}!`, oldSettings)
-            await sbp('gi.db/settings/delete', this.form.name)
-          }
-          // proceed with creation
-          const user = sbp('gi.contracts/identity/create', {
-            // authorizations: [Events.CanModifyAuths.dummyAuth()],
-            attributes: {
-              name: this.form.name,
-              email: this.form.email,
-              picture: `${window.location.origin}/assets/images/default-avatar.png`
-            }
-          })
-          const mailbox = sbp('gi.contracts/mailbox/create', {
-            // authorizations: [Events.CanModifyAuths.dummyAuth(user.hash())]
-          })
-          await sbp('backend/publishLogEntry', user)
-          await sbp('backend/publishLogEntry', mailbox)
-
-          // set the attribute *after* publishing the identity contract
-          const attribute = await sbp('gi.contracts/identity/setAttributes/create',
-            { mailbox: mailbox.hash() },
-            user.hash()
-          )
-          await sbp('backend/publishLogEntry', attribute)
-          // register our username if contract creation worked out
-          await sbp('namespace/register', this.form.name, user.hash())
-          // call syncContractWithServer on all of these contracts to:
-          // 1. begin monitoring the contracts for updates via the pubsub system
-          // 2. add these contracts to our vuex state
-          for (const contract of [user, mailbox]) {
-            await sbp('state/enqueueContractSync', contract.hash())
-          }
-          // TODO: Just add cryptographic magic
-          // login also calls 'state/enqueueContractSync', but not in this case since we
-          // just sync'd it.
-          await sbp('state/vuex/dispatch', 'login', {
-            username: this.form.name,
-            identityContractID: user.hash()
-          })
-          this.ephemeral.errorMsg = 'success' // TODO: get rid of this and fix/update tests accordingly
-          if (this.$route.query.next) {
-            // TODO: get rid of this timeout and fix/update tests accordingly
-            setTimeout(() => {
-              this.$router.push({ path: this.$route.query.next })
-            }, 1000)
-          } else {
-            this.$router.push({ path: '/' })
-          }
-          this.$refs.modal.close()
-        } catch (ex) {
-          console.error('SignUp.vue submit() error:', ex)
-          sbp('state/vuex/dispatch', 'logout')
-          this.ephemeral.errorMsg = ex.toString()
-        }
+    submit () {
+      if (this.$route.query.next) {
+        // TODO: get rid of this timeout and fix/update tests accordingly
```

Comment:
@taoeffect do you know why exactly we had to add this timeout here? I believe this was an issue from nightmare times... not cypress :D